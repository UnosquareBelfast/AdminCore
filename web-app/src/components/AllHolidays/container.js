import React, { Component } from 'react';
import { getAllHolidays } from '../../services/holidayService';

export default Wrapped =>
  class extends Component {
    constructor(props) {
      super(props);
      this.state = { holidays: [], selectedHoliday: {} };
    }

    componentDidMount() {
      this.getHolidays();
    }

    getHolidays = () => {
      getAllHolidays().then(response => {
        const holidays = response.data;
        this.setState({ holidays });
      });
    };

    selectHoliday = holiday => this.setState({ selectedHoliday: holiday });

    closeModal = () => {
      this.selectHoliday({});
      this.getHolidays();
    };

    render() {
      return (
        <Wrapped
          {...this.props}
          holidays={this.state.holidays}
          selectedHoliday={this.state.selectedHoliday}
          selectHoliday={this.selectHoliday}
          closeModal={this.closeModal}
        />
      );
    }
  };
