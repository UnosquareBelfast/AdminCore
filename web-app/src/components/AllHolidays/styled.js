import styled from 'styled-components';

export const HolidayTable = styled.table`
  width: 100%;

  tr:nth-child(even) {
    background-color: ${props => props.theme.colours.lightgrey};
  }

  tr td:last-of-type {
    text-align: right;
  }

  th {
    text-align: left;
  }

  button {
    border: none;
    background: ${props => props.theme.colours.unoBlue};
    color: white;
    padding: 3px 10px;
    margin: 2px 5px 2px 0;
    border-radius: 3px;
    cursor: pointer;
  }

  button:hover {
    background: ${props => props.theme.colours.darkBlue};
  }
`;

export const StatusDot = styled.div`
  display: inline-block;
  position: relative;
  top: -2.5px;
  margin-right: 7px;
  height: 7px;
  width: 7px;
  border-radius: 7px;
  background: ${({ theme, status }) => theme.holidayStatus[status]};
`;
