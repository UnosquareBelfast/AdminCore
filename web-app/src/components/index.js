export { default as AdminDashboard } from './AdminDashboard';
export { default as AllClients } from './AllClients';
export { default as AllHolidays } from './AllHolidays';
export { default as BigCalendarToolbar } from './BigCalendarToolbar';
export { default as BookedLeave } from './BookedLeave';
export { default as BookingCalendar } from './BookingCalendar';
export { default as BookingModal } from './BookingModal';
export { default as CreateClient } from './CreateClient';
export { default as CreateContract } from './CreateContract';
export { default as ClientModal } from './ClientModal';
export { default as CreateTeam } from './CreateTeam';
export { default as CreateUser } from './CreateUser';
export { default as Legend } from './Legend';
export { default as PendingHolidays } from './PendingHolidays';
export { default as RangeDayPicker } from './common/RangeDayPicker';
export { default as RequestHoliday } from './RequestHoliday';
export { default as TakenLeave } from './TakenLeave';
export { default as TeamView } from './TeamView';
export { default as UserDetails } from './UserDetails';
export { default as UserModal } from './UserModal';
export { default as ViewContracts } from './ViewContracts';
export { default as ViewTeams } from './ViewTeams';
export { default as HolidayModal } from './HolidayModal';
export { default as AllEmployees } from './AllEmployees';
export { default as DataTable } from './DataTable';
