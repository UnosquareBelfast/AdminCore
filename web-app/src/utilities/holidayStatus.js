import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import {
  faThumbsUp,
  faThumbsDown,
  faClock,
  faSun,
} from '@fortawesome/fontawesome-free-solid';

export default {
  PENDING: 1,
  APPROVED: 2,
  REJECTED: 3,
  MANDATORY: 4,
};

export const statusText = [
  null,
  'Pending',
  'Approved',
  'Rejected',
  'Mandatory',
];

export const statusIcons = [
  null,
  <FontAwesomeIcon icon={faClock} />,
  <FontAwesomeIcon icon={faThumbsUp} />,
  <FontAwesomeIcon icon={faThumbsDown} />,
  <FontAwesomeIcon icon={faSun} />,
];
