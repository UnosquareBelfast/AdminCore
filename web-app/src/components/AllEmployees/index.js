import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import { UserList, UserModal } from '../';
import { Button } from '../common';
import { CornerButton } from '../common_styled';
import container from './container';

const AllEmployees = ({ history, users, selectUser, selectedUser }) => {
  return (
    <Fragment>
      <CornerButton>
        <Button
          onClick={() => history.replace('/admin/employees/new')}
          label="New Employee"
        />
      </CornerButton>
      <UserModal
        user={selectedUser}
        closeModal={() => selectUser({})}
        history={history}
      />
      <h2>Employees</h2>
      <UserList
        history={history}
        users={users}
        columns={['fullName', 'email', 'role', 'location']}
        onRowClick={employee => selectUser(employee)}
      />
    </Fragment>
  );
};

AllEmployees.propTypes = {
  history: PT.object.isRequired,
  users: PT.array.isRequired,
  selectUser: PT.func.isRequired,
  selectedUser: PT.object.isRequired,
};

export default container(AllEmployees);
