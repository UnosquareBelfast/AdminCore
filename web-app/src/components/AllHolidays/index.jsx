import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { HolidayList, HolidayModal } from '../';

export const AllHolidays = ({ holidays, selectedHoliday, selectHoliday }) => {
  return (
    <Fragment>
      <HolidayModal
        holiday={selectedHoliday}
        closeModal={() => selectHoliday({})}
      />
      <h2>All Holidays</h2>
      <HolidayList
        holidays={holidays}
        columns={[
          'status',
          'employee',
          'startDate',
          'endDate',
          'requestedDate',
        ]}
        onRowClick={holiday => selectHoliday(holiday)}
      />
    </Fragment>
  );
};

AllHolidays.propTypes = {
  holidays: PT.array.isRequired,
  selectedHoliday: PT.object.isRequired,
  selectHoliday: PT.func.isRequired,
};

export default container(AllHolidays);
