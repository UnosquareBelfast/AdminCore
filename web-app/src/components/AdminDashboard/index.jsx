import React from 'react';
import { Link } from 'react-router-dom';
import { InnerLayout } from './styled';

const AdminDashboard = () => {
  return (
    <InnerLayout>
      <h2>Admin Dashboard</h2>
      <div className="columnWrap">
        <div>
          <h3>Employees</h3>
          <Link to="/admin/employees">View Employees</Link>
        </div>
        <div>
          <h3>Holidays</h3>
          <Link to="/admin/holidays">View All Holidays</Link>
          <Link to="/admin/pendingHolidays">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Clients</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Contracts</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
      </div>
    </InnerLayout>
  );
};

export default AdminDashboard;
