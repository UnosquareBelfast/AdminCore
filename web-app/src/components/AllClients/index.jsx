import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { Table, Button } from '../common';
import { ActiveDot, CTA } from './styled';
import employeeStatus from '../../utilities/employeeStatus';

export const AllClients = ({
  clients,
  viewClient,
  archive,
  createNewClient,
}) => {
  let clientList;
  if (clients.length > 0) {
    clientList = clients.map(client => (
      <tr key={client.clientId}>
        <td>{client.clientName}</td>
        <td>{client.clientStatusDescription}</td>
        <td>
          <ActiveDot active={client.clientStatusId === employeeStatus.ACTIVE} />
          {client.clientStatusId === employeeStatus.ACTIVE
            ? 'Active'
            : 'Inactive'}
        </td>
        <td>
          <button onClick={() => viewClient(client.clientId)}>
            Edit Client
          </button>
          {client.clientStatusId === employeeStatus.ACTIVE ? (
            <button className="success" onClick={() => archive(client)}>
              Archive
            </button>
          ) : (
            <button className="error" onClick={() => archive(client, true)}>
              Activate
            </button>
          )}
        </td>
      </tr>
    ));
  } else {
    clientList = (
      <tr>
        <td>No Clients</td>
        <td />
        <td />
        <td />
      </tr>
    );
  }
  return (
    <Fragment>
      <h2>All Clients</h2>
      <Table
        tableHeaders={['Name', 'Description', 'Status', ' ']}
        tableRows={clientList}
      />
      <CTA>
        <Button onClick={() => createNewClient()} label="New Client" />
      </CTA>
    </Fragment>
  );
};

AllClients.propTypes = {
  createNewClient: PT.func,
  clients: PT.array,
  archive: PT.func.isRequired,
  viewClient: PT.func.isRequired,
};

export default container(AllClients);
