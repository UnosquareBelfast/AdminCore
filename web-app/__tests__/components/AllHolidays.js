import React from 'react';
import { shallow } from 'enzyme';
import { AllHolidays } from '../../src/components/AllHolidays';
import holidayStatus from '../../src/utilities/holidayStatus';

const dummyHolidays = [
  {
    employee: {
      forename: 'Joe',
      surname: 'Bloggs',
    },
    holidayId: 123,
    date: [1, 1, 2018],
    dateCreated: new Date(),
    holidayStatusId: holidayStatus.PENDING,
  },
];
const closeModal = jest.fn();
const selectHoliday = jest.fn();

describe('AllHolidays', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AllHolidays
        holidays={dummyHolidays}
        selectedHoliday={{}}
        selectHoliday={selectHoliday}
        closeModal={closeModal}
      />,
    );
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
