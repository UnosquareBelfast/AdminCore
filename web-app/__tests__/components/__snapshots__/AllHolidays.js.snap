// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllHolidays renders correctly 1`] = `
<Card>
  <h3>
    All Holidays
  </h3>
  <styled.table>
    <tbody>
      <tr>
        <th>
          Status
        </th>
        <th>
          Employee
        </th>
        <th>
          Date
        </th>
        <th>
          Created
        </th>
      </tr>
      <tr
        key="123"
      >
        <td>
          <styled.div
            status={1}
          />
          Pending
        </td>
        <td>
          Joe Bloggs
        </td>
        <td>
          2018/1/1
        </td>
        <td>
          undefined/undefined/undefined
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </styled.table>
</Card>
`;
